### PR TITLE
Refactor skills

### PR DESCRIPTION
**Pull request by johnnypowa**
### Refactor skills

-AppDatabase version change 1 to 2
-Create typeConverter to skillList
-Add skills columnInfo to KnightEntity
-Add skillList to Knight
-Create skills in CombatLocalDataSource
-Add skills to skillList in CombatLocalDataSource
-Add skillList to KnightDao.insertKnight in CombatLocalDataSource
-Create skillActivation function and checkEquippedSkills function, remove old skill functions on CombatFragment
-Use skillActivation function on knightSkills setOnClickListeners
-Add skillList to knightEntityData on CombatViewModel
-Create Skill class

### Plan

-Show the equipped skills icons and names on the CombatFragment
-Create RecyclerView layout for skills 
-Users can equip or unequip the skills on the skills fragment. Max. 4 skills can be equipped.

### CODE REVIEW

File: app/src/main/java/com/example/tanuls2/ui/view/CombatFragment.kt [hunk 1]
```diff
@@ -311,51 +316,57 @@ class CombatFragment : Fragment() {
...
-    fun skillEliminateBlockChance() {
-        clearZombieHitResults()
-        val originalBlockChance = zombie.blockChance
-        zombie.blockChance = 0.0f
-        hitWithKnight()
-        zombie.blockChance = originalBlockChance
-        skillDisable(knightSkill3Id)
-        printKnightParameter(combatViewModel.currentKnight!!)
-        printZombieParameter(zombie)
+    fun checkEquippedSkills(){
```
Comment (csepidave) [inline app/src/main/java/com/example/tanuls2/ui/view/CombatFragment.kt:319]: let's talk about it :speech_balloon:

File: app/src/main/java/com/example/tanuls2/ui/view/CombatFragment.kt [hunk 2]
```diff
@@ -311,51 +316,57 @@ class CombatFragment : Fragment() {
...
+        equippedSkillsList = arrayListOf()
+        for (i in 0..skillListLength-1) {
+            if(combatViewModel.currentKnight!!.skillList[i].equipped){
+                equippedSkillsList.add(combatViewModel.currentKnight!!.skillList[i])
+            }
+        }
     }
 
-    fun skillBloodSiphon() {
+    fun skillActivation(index : Int){
```
Comment (csepidave) [inline app/src/main/java/com/example/tanuls2/ui/view/CombatFragment.kt:324]: let's talk about it :speech_balloon:

File: app/src/main/java/com/example/tanuls2/service/datasource/CombatLocalDataSource.kt [hunk 3]
```diff
@@ -3,24 +3,29 @@ package com.example.tanuls2.service.datasource
...
+    var skillPrecisionHit = Skill("Precíz ütés",1,0,0,0f,0.0f,0f,true)
+    var skillCriticalHit  = Skill ("Kritikus ütés",1,0,0,1.0f,0f,0f,true)
+    var skillLifeSteal  = Skill("Gyógyító ütés", 1,0,0,0f,0f,0.5f,true)
+    var skillHeal = Skill("Gyógyítás", 1,25,0,0f,0f,0f,false)
+    
     fun fetchLocalKnightData() : Single<KnightEntity> {
         return Single.fromCallable {
             if(SharedPreferencesHandler.isFirstStart()){
-                var itemList = arrayListOf<Item>()
+                var skillList = arrayListOf(skillExtraDamage,skillCriticalHit,skillPrecisionHit,skillLifeSteal,skillHeal)
```
Comment (csepidave) [inline app/src/main/java/com/example/tanuls2/service/datasource/CombatLocalDataSource.kt]: segmentation!
```
arrayListOf(skillExtraDamage, skillCriticalHit, skillPrecisionHit, skillLifeSteal, skillHeal)
```

File: app/src/main/java/com/example/tanuls2/service/datasource/CombatLocalDataSource.kt [hunk 4]
```diff
@@ -3,24 +3,29 @@ package com.example.tanuls2.service.datasource
...
-import com.example.tanuls2.model.Knight
-import com.example.tanuls2.model.Zombie
+import com.example.tanuls2.model.*
 import io.reactivex.rxjava3.core.Completable
 import io.reactivex.rxjava3.core.Single
 import io.reactivex.rxjava3.schedulers.Schedulers
 
 class CombatLocalDataSource(val knightDao: KnightDao) {
 
+    var skillExtraDamage = Skill("Dupla ütés", 1,0,100,0f,0f,0f,true)
```
Comment (csepidave) [inline app/src/main/java/com/example/tanuls2/service/datasource/CombatLocalDataSource.kt]: Please pay attention to segmentation!
```
Skill("Dupla ütés", 1, 0, 100, 0f, 0f , 0f, true)
```

File: app/src/main/java/com/example/tanuls2/model/Skill.kt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package com.example.tanuls2.model
+
+open class Skill(var skillName:String?,
+                 var level:Int,
+                 var health:Int,
+                 var damage:Int,
+                 var criticalHitChance: Float,
```
Comment (csepidave) [inline app/src/main/java/com/example/tanuls2/model/Skill.kt:7]: Use the same segmentation like this on the other lines too.

File: app/src/main/java/com/example/tanuls2/ui/view/CombatFragment.kt [hunk 6]
```diff
@@ -311,51 +316,57 @@ class CombatFragment : Fragment() {
...
         clearZombieHitResults()
-        val actualHealthOfEnemy = zombie.currentHealth
-        hitWithKnight()
-        val modifiedHealth =
-            combatViewModel.currentKnight!!.currentHealth + (actualHealthOfEnemy - zombie.currentHealth)
-        combatViewModel.currentKnight!!.currentHealth = when (modifiedHealth >= combatViewModel.currentKnight!!.maxHealth) {
-            true -> {
-                combatViewModel.currentKnight!!.maxHealth
+        when(equippedSkillsList[index].skillName){
+            getString(R.string.skill_double_hit)->{
```
Comment (csepidave) [inline app/src/main/java/com/example/tanuls2/ui/view/CombatFragment.kt]: segmentation

**Review by csepidave — CHANGES_REQUESTED** [review]
There are some small formatting issues and issues which we have to discuss.
But overall nice work! 👍

File: app/src/main/java/com/example/tanuls2/service/datasource/CombatLocalDataSource.kt (hunk 4 above)
Comment (johnnypowa) [reply app/src/main/java/com/example/tanuls2/service/datasource/CombatLocalDataSource.kt]: Done

File: app/src/main/java/com/example/tanuls2/service/datasource/CombatLocalDataSource.kt (hunk 3 above)
Comment (johnnypowa) [reply app/src/main/java/com/example/tanuls2/service/datasource/CombatLocalDataSource.kt]: Done

File: app/src/main/java/com/example/tanuls2/model/Skill.kt (hunk 5 above)
Comment (johnnypowa) [reply app/src/main/java/com/example/tanuls2/model/Skill.kt:7]: Done

File: app/src/main/java/com/example/tanuls2/ui/view/CombatFragment.kt (hunk 6 above)
Comment (johnnypowa) [reply app/src/main/java/com/example/tanuls2/ui/view/CombatFragment.kt]: Done

File: app/src/main/java/com/example/tanuls2/ui/view/CombatFragment.kt (hunk 1 above)
Comment (johnnypowa) [reply app/src/main/java/com/example/tanuls2/ui/view/CombatFragment.kt:319]: Done

File: app/src/main/java/com/example/tanuls2/ui/view/CombatFragment.kt (hunk 2 above)
Comment (johnnypowa) [reply app/src/main/java/com/example/tanuls2/ui/view/CombatFragment.kt:324]: Done

**Review by csepidave — APPROVED** [review]

**Pull request merged**